internal/client/garden/client.go

```diff
@@ -472,30 +473,73 @@ func (g *clientImpl) GetSeedClientConfig(ctx context.Context, name string) (clie
 	return config, nil
 }
 
-func (g *clientImpl) GetCloudProfile(ctx context.Context, name string) (*gardencorev1beta1.CloudProfile, error) {
-	cloudProfile := &gardencorev1beta1.CloudProfile{}
-	key := types.NamespacedName{Name: name}
+func (g *clientImpl) GetCloudProfile(ctx context.Context, ref gardencorev1beta1.CloudProfileReference) (*AbstractCloudProfile, error) {
+	switch ref.Kind {
+	case corev1beta1constants.CloudProfileReferenceKindCloudProfile:
+		cloudProfile := &gardencorev1beta1.CloudProfile{}
+		key := types.NamespacedName{Name: ref.Name}
 
-	if err := g.c.Get(ctx, key, cloudProfile); err != nil {
-		return nil, fmt.Errorf("failed to get cloudprofile %v: %w", key, err)
-	}
+		if err := g.c.Get(ctx, key, cloudProfile); err != nil {
+			return nil, fmt.Errorf("failed to get CloudProfile %v: %w", key, err)
+		}
+
+		return &AbstractCloudProfile{
+			CloudProfile: cloudProfile,
+		}, nil
+
+	case corev1beta1constants.CloudProfileReferenceKindNamespacedCloudProfile:
+		namespacedCloudProfile := &gardencorev1beta1.NamespacedCloudProfile{}
+		key := types.NamespacedName{Name: ref.Name}
 
-	return cloudProfile, nil
+		if err := g.c.Get(ctx, key, namespacedCloudProfile); err != nil {
+			return nil, fmt.Errorf("failed to get NamespacedCloudProfile %v: %w", key, err)
+		}
+
+		return &AbstractCloudProfile{
+			NamespacedCloudProfile: namespacedCloudProfile,
+		}, nil
+
+	default:
+		return nil, fmt.Errorf("unknown CloudProfile kind: %s", ref.Kind)
+	}
 }
 
 // RuntimeClient returns the underlying Kubernetes runtime client.
 func (g *clientImpl) RuntimeClient() client.Client {
 	return g.c
 }
 
-type CloudProfile gardencorev1beta1.CloudProfile
+// AbstractCloudProfile encapsulates a CloudProfile or NamespacedCloudProfile.
+type AbstractCloudProfile struct {
```

Comment:
As discussed rename to `CloudProfileUnion`

Reply:
✅